cfme/tests/openstack/cloud/test_networks.py

```diff
@@ -0,0 +1,216 @@
+
+import fauxfactory
+import pytest
+
+from cfme.cloud.provider.openstack import OpenStackProvider
+from cfme.utils import testgen
+from cfme.utils.appliance.implementations.ui import navigate_to
+from cfme.utils.log import logger
+from cfme.utils.wait import wait_for
+
+
+pytest_generate_tests = testgen.generate([OpenStackProvider],
+                                         scope='module')
+
+pytestmark = [pytest.mark.usefixtures("setup_provider_modscope")]
+
+
+def delete_entity(entity):
+    # TODO: replace this with neutron client request
+    try:
+        if entity.exists:
+            entity.delete()
+    except Exception:
+        logger.warning('Exception during network entity deletion - skipping..')
+
+
+def create_network(appliance, provider, is_external):
+    collection = appliance.collections.cloud_networks
+    network = collection.create(name=fauxfactory.gen_alpha(),
+                                tenant=provider.get_yaml_data()['tenant'],
+                                provider=provider,
+                                network_type='VXLAN',
+                                network_manager='{} Network Manager'.format(provider.name),
+                                is_external=is_external)
+    return network
+
+
+def create_subnet(appliance, provider, network, cidr):
+    collection = appliance.collections.network_subnets
+    subnet = collection.create(name=fauxfactory.gen_alpha(),
+                               tenant=provider.get_yaml_data()['tenant'],
+                               provider=provider,
+                               network_manager='{} Network Manager'.format(provider.name),
+                               network_name=network.name,
+                               cidr=cidr)
+    return subnet
+
+
+def create_router(appliance, provider, ext_gw, ext_network=None, ext_subnet=None):
+    collection = appliance.collections.network_routers
+    router = collection.create(name=fauxfactory.gen_alpha(),
+                               tenant=provider.get_yaml_data()['tenant'],
+                               provider=provider,
+                               network_manager='{} Network Manager'.format(provider.name),
+                               has_external_gw=ext_gw,
+                               ext_network=ext_network,
+                               ext_network_subnet=ext_subnet)
+    return router
+
+
+@pytest.yield_fixture(scope='function')
+def network(provider, appliance):
+    """Create cloud network"""
+    network = create_network(appliance, provider, is_external=False)
+    yield network
+    delete_entity(network)
+
+
+@pytest.yield_fixture(scope='function')
+def ext_network(provider, appliance):
+    """Create external cloud network"""
+    network = create_network(appliance, provider, is_external=True)
+    yield network
+    delete_entity(network)
+
+
+@pytest.fixture(scope='module')
+def subnet_cidr():
+    return '11.11.11.0/24'
```

Comment:
**Question:** We use the same value for every test?

Reply:
Yes..